example/object.h

```diff
@@ -55,37 +62,43 @@ class Object {
 template <typename T> class ref {
 public:
 	/// Create a nullptr reference
-    ref() : m_ptr(nullptr) { std::cout << "Created empty ref" << std::endl; }
+    ref() : m_ptr(nullptr) { print_default_created(this); track_default_created((ref_tag*) this); }
```

Comment:
Correct me if I'm wrong: the semantics here seem to be a little different: the prior ``cout`` calls printed information about the lifetime of an instance which is managed via the ``ref<>`` wrapper. Here, you are actually tracking ``this``, i.e. the memory address of the individual ``ref<>`` wrappers which is less relevant.